bindings/c/test/mako/async.cpp

```diff
@@ -268,13 +237,20 @@ void ResumableStateForRunWorkload::onTransactionSuccess() {
 		stats.incrOpCount(OP_TRANSACTION);
 		watch_tx.startFromStop();
 		tx.reset();
-		if (ended()) {
-			signalEnd();
-		} else {
-			iter = getOpBegin(args);
-			// start next iteration
-			postNextTick();
-		}
+		restartIteration(FutureRC::OK);
+	}
+}
+void ResumableStateForRunWorkload::restartIteration(FutureRC rc) {
```

Comment:
The name is declarative as if it always restarts, but it doesn't so the name is misleading.